pageserver/src/layered_repository.rs

```diff
@@ -781,6 +738,12 @@ pub struct LayeredTimeline {
     // Needed to ensure that we can't create a branch at a point that was already garbage collected
     latest_gc_cutoff_lsn: RwLock<Lsn>,
 
+    // List of child timelines and their branch points. This is needed to avoid
+    // garbage collecting data that is still needed by the child timelines.
+    gc_info: RwLock<GcInfo>,
+
+    partitioning: RwLock<Option<(KeyPartitioning, Lsn)>>,
```

Comment:
Docstring would be very useful. Hard to infer what the `Vec<Vec<_>>` represents, especially since the `KeyPartitioning` docstring seems outdated.